pymc/util.py

```diff
@@ -607,6 +622,257 @@ def update(
+        """Get a table to render the Progress display.
+
+        Unlike the parent method, this one returns a full table (not a grid), allowing for column headings.
+
+        Parameters
+        ----------
+        tasks: Iterable[Task]
+            An iterable of Task instances, one per row of the table.
+
+        Returns
+        -------
+        table: Table
+            A table instance.
+        """
+
+        def call_column(column, task):
+            if hasattr(column, "callbacks"):
+                column.callbacks(task)
+
+            return column(task)
+
+        table_columns = (
+            (
+                Column(no_wrap=True)
+                if isinstance(_column, str)
+                else _column.get_table_column().copy()
+            )
+            for _column in self.columns
+        )
+        if self.include_headers:
+            table = Table(
+                *table_columns,
+                padding=(0, 1),
+                expand=self.expand,
+                show_header=True,
+                show_edge=True,
+                box=SIMPLE_HEAD,
+            )
+        else:
+            table = Table.grid(*table_columns, padding=(0, 1), expand=self.expand)
+
+        for task in tasks:
+            if task.visible:
+                table.add_row(
+                    *(
+                        (
+                            column.format(task=task)
+                            if isinstance(column, str)
+                            else call_column(column, task)
+                        )
+                        for column in self.columns
+                    )
+                )
+
+        return table
+
+
+class DivergenceBarColumn(BarColumn):
+    def __init__(self, *args, diverging_color="red", **kwargs):
+        from matplotlib.colors import to_rgb
+
+        self.diverging_color = diverging_color
+        self.diverging_rgb = [int(x * 255) for x in to_rgb(self.diverging_color)]
+
+        super().__init__(*args, **kwargs)
+
+        self.non_diverging_style = self.complete_style
+        self.non_diverging_finished_style = self.finished_style
+
+    def callbacks(self, task: "Task"):
+        divergences = task.fields.get("divergences", 0)
+        if isinstance(divergences, float | int) and divergences > 0:
+            self.complete_style = Style.parse("rgb({},{},{})".format(*self.diverging_rgb))
+            self.finished_style = Style.parse("rgb({},{},{})".format(*self.diverging_rgb))
+        else:
+            self.complete_style = self.non_diverging_style
+            self.finished_style = self.non_diverging_finished_style
+
+
+class ProgressManager:
+    def __init__(self, step_method, chains, draws, tune, progressbar, progressbar_theme):
+        mode = "chain"
+        stats = "full"
+
+        if isinstance(progressbar, bool):
+            show_progress = progressbar
+        else:
+            show_progress = True
+
+            if "+" in progressbar:
```

Comment:
Perhaps more readable / less bug-prone to if/elif all the valid cases?

Reply:
Yeah definitely. Or replace `progress_theme` with `progress_kwargs` that takes `compact: bool, show_fewer_stats: bool, progress_theme: whatever` ?

Reply:
I think that's harder to use